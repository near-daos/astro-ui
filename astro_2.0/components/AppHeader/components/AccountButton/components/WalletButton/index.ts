export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletButton/WalletButton';
