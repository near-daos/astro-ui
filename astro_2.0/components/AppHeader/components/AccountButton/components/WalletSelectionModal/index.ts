export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletSelectionModal/WalletSelectionModal';
