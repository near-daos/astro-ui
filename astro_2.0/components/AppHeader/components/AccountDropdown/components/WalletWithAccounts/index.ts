export * from 'astro_2.0/components/AppHeader/components/AccountDropdown/components/WalletWithAccounts/WalletWithAccounts';
