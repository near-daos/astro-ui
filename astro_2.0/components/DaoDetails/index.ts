export * from './DaoDetailsMinimized';
export * from './DaoDetailsGrid';
