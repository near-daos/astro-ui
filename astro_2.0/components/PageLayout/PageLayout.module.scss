@import 'styles/mixins';

.root {
  background-color: var(--color-neutral-10);
  display: flex;
  position: relative;
  width: 100%;

  &.whiteBackground {
    background-color: var(--color-white);
  }
}

.main {
  $header-height: 72px;

  padding: 0 24px 60px;
  width: 100%;

  @include desktop {
    padding: 0 48px 60px;
  }

  @include large-desktop {
    margin: auto;
    max-width: 1000px;
    padding: 0 60px 60px;
  }

  @include large-x-desktop {
    margin: auto;
    max-width: 1125px;
    padding: 0 94px 60px;
  }
}

.content {
  width: 100%;

  &.withFixedHeader {
    padding-top: var(--app-header-height);
  }
}

.createDao.root {
  padding-bottom: 80px;

  @include astro-desktop {
    padding-bottom: 0;
  }

  .main {
    padding: 24px 24px 60px;

    @include desktop {
      padding: 50px 48px 60px;
    }

    @include large-desktop {
      margin: auto;
      max-width: 1180px;
      padding: 50px 72px 60px;
    }
  }
}
