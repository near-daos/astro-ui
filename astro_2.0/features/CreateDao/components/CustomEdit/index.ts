import CustomEdit from './CustomEdit';

export default CustomEdit;
