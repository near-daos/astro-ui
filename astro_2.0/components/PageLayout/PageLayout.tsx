import cn from 'classnames';
import { useRouter } from 'next/router';
import React, { FC } from 'react';
import Tooltip from 'react-tooltip';
import { useFlags } from 'launchdarkly-react-client-sdk';

import { CREATE_DAO_URL } from 'constants/routing';

import { Sidebar as SidebarNext } from 'astro_3.0/features/Sidebar';
import { Sidebar } from 'components/Sidebar';
import { AppHeader } from 'astro_2.0/components/AppHeader';
import { AppHeader as AppHeaderNext } from 'astro_3.0/features/AppHeader';
import { LinkToTop } from 'astro_2.0/components/LinkToTop';
import { NotificationContainer } from 'features/notifications';
import { MaintenanceWarning } from 'astro_2.0/components/MaintenanceWarning';

import styles from './PageLayout.module.scss';

interface Props {
  appVersion: number | null;
}

export const PageLayout: FC<Props> = ({ children, appVersion }) => {
  const router = useRouter();
  const { applicationMaintenance } = useFlags();

  const isCreateDaoPage = router.route.match(CREATE_DAO_URL);

  const rootClassName = cn(styles.root, {
    [styles.createDao]: isCreateDaoPage,
    [styles.whiteBackground]: appVersion === 3,
  });

  return (
    <div className={rootClassName}>
      {appVersion === 3 ? <SidebarNext /> : <Sidebar />}
      <div
        className={cn(styles.content, {
          [styles.withFixedHeader]: appVersion === 3,
        })}
      >
        {appVersion === 3 ? <AppHeaderNext /> : <AppHeader />}
        {applicationMaintenance ? <MaintenanceWarning /> : children}
        <LinkToTop />
      </div>
      <NotificationContainer />
      <Tooltip effect="solid" />
    </div>
  );
};
