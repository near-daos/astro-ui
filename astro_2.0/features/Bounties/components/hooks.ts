import { useCallback, useRef, useState } from 'react';
import { useRouter } from 'next/router';
import { useAsyncFn } from 'react-use';
import axios, { CancelTokenSource } from 'axios';

import { SputnikHttpService } from 'services/sputnik';
import { useModal } from 'components/modal';
import { ConfirmActionModal } from 'astro_2.0/components/ConfirmActionModal';

import { DAO } from 'types/dao';
import { Bounty, BountyContext, BountyProposal } from 'types/bounties';
import { VoteAction } from 'types/proposal';

import { NOTIFICATION_TYPES, showNotification } from 'features/notifications';

import { useAuthContext } from 'context/AuthContext';
import { PaginationResponse } from 'types/api';

export function useBountyControls(
  dao: DAO,
  bounty?: Bounty
): {
  handleClaim: () => void;
  handleUnclaim: () => void;
} {
  const router = useRouter();
  const { nearService } = useAuthContext();

  const [showModal] = useModal(ConfirmActionModal);

  const onSuccessHandler = useCallback(async () => {
    await router.reload();
  }, [router]);

  const handleClaim = useCallback(async () => {
    const res = await showModal({
      title: 'Confirm Your Claim',
    });

    if (res?.length && bounty) {
      await nearService?.claimBounty(dao.id, {
        bountyId: bounty?.bountyId,
        deadline: bounty?.maxDeadline,
        bountyBond: dao.policy.bountyBond,
        gas: res[0],
        tokenId: bounty?.token,
      });

      onSuccessHandler();
    }
  }, [
    showModal,
    bounty,
    dao.id,
    dao.policy.bountyBond,
    onSuccessHandler,
    nearService,
  ]);

  const handleUnclaim = useCallback(async () => {
    const res = await showModal({
      title: 'Confirm Your Unclaim',
    });

    if (res?.length && bounty) {
      await nearService?.unclaimBounty(dao.id, bounty?.bountyId);
      onSuccessHandler();
    }
  }, [bounty, dao.id, onSuccessHandler, showModal, nearService]);

  return {
    handleClaim,
    handleUnclaim,
  };
}

export function useBountyVoting(
  dao: DAO,
  proposal: BountyProposal
): {
  handleVote: (vote: VoteAction) => void;
  loading: boolean;
} {
  const router = useRouter();
  const [showModal] = useModal(ConfirmActionModal);
  const { nearService } = useAuthContext();

  const [{ loading }, handleVote] = useAsyncFn(
    async (vote: VoteAction) => {
      const res = await showModal({
        title: 'Confirm Your Vote',
      });

      if (res?.length) {
        await nearService?.vote(dao.id, proposal.proposalId, vote, res[0]);
        await router.reload();
      }
    },
    [dao, proposal, router]
  );

  return {
    handleVote,
    loading,
  };
}

export function useBountySearch(): {
  handleSearch: (
    val: string
  ) => Promise<PaginationResponse<BountyContext[]> | null>;
  loading: boolean;
} {
  const router = useRouter();
  const daoId = router.query.dao as string;
  const { accountId } = useAuthContext();
  const cancelTokenRef = useRef<CancelTokenSource | null>(null);

  const [{ loading }, handleSearch] = useAsyncFn(async query => {
    if (cancelTokenRef.current) {
      cancelTokenRef.current?.cancel('Cancelled by new req');
    }

    const { CancelToken } = axios;
    const source = CancelToken.source();

    cancelTokenRef.current = source;

    return SputnikHttpService.findBountyContext({
      daoId,
      accountId,
      query,
      cancelToken: source.token,
    });
  }, []);

  return {
    handleSearch,
    loading,
  };
}

export function useHideBounty(): {
  handleSubmit: () => void;
  handleSelect: (id: string) => void;
  selected: string[];
  loading: boolean;
} {
  const router = useRouter();
  const showHidden = router.query?.bountyFilter === 'hidden';
  const daoId = router.query.dao as string;
  const { accountId, nearService } = useAuthContext();
  const [selected, setSelected] = useState<string[]>([]);

  const [{ loading }, handleSubmit] = useAsyncFn(async () => {
    try {
      const publicKey = await nearService?.getPublicKey();
      const signature = await nearService?.getSignature();

      if (publicKey && signature) {
        await SputnikHttpService.toggleBountyContexts({
          accountId,
          publicKey,
          signature,
          daoId,
          ids: selected,
          isArchived: !showHidden,
        });

        await router.replace(router.asPath, undefined, {
          shallow: false,
        });

        setSelected([]);
      }
    } catch (err) {
      showNotification({
        type: NOTIFICATION_TYPES.ERROR,
        lifetime: 20000,
        description: err.message,
      });
    }
  }, [selected, daoId, showHidden, accountId, router]);

  const handleSelect = useCallback(
    (id: string) => {
      if (selected.includes(id)) {
        setSelected(selected.filter(item => item !== id));
      } else {
        setSelected([...selected, id]);
      }
    },
    [selected]
  );

  return {
    handleSubmit,
    handleSelect,
    selected,
    loading,
  };
}
