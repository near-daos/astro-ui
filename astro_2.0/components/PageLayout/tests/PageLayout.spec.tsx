import { render } from 'jest/testUtils';

import { PageLayout } from 'astro_2.0/components/PageLayout';

jest.mock('features/notifications', () => {
  return {
    NotificationContainer: () => 'NotificationContainer',
    useNotifications: jest.fn(() => ({})),
  };
});

describe('page layout', () => {
  it('Should render component', () => {
    const content = 'Hello World!';

    const { getByText } = render(
      <PageLayout appVersion={2}>{content}</PageLayout>
    );

    expect(getByText(content)).toBeInTheDocument();
  });
});
