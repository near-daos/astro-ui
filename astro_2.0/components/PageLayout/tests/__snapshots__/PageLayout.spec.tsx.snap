// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page layout Should render component 1`] = `
<div>
  <div
    class=""
  >
    <aside
      class=""
    >
      <div>
        <a
          class=""
          href="https://astrodao.com/"
        >
          <svg
            width="120"
          >
            <use
              href="/_next/undefined"
            />
          </svg>
        </a>
        <div>
          <span>
            powered by
          </span>
          <i>
            <svg
              width="44"
            >
              <use
                href="/_next/undefined"
              />
            </svg>
          </i>
        </div>
        <div>
          <div>
            <a
              class=""
              href="/all/daos"
            >
              <svg
                height="24"
              >
                <use
                  href="/_next/undefined"
                />
              </svg>
              <span>
                 
                allDaos
                 
              </span>
            </a>
            <a
              class=""
              href="/discover"
            >
              <svg
                height="24"
              >
                <use
                  href="/_next/undefined"
                />
              </svg>
              <span>
                 
                discover.title
                 
              </span>
            </a>
            <a
              class=""
              href="/all/feed"
            >
              <svg
                height="24"
              >
                <use
                  href="/_next/undefined"
                />
              </svg>
              <span>
                 
                globalFeed
                 
              </span>
            </a>
            <div />
          </div>
          <div
            class=""
          >
            <svg
              height="24"
            >
              <use
                href="/_next/undefined"
              />
            </svg>
            <span>
               
              createNewDao
               
            </span>
          </div>
        </div>
      </div>
      <div
        class=""
        role="button"
        tabindex="0"
      >
        <div>
          <div>
            <div>
              <a
                href="https://twitter.com/AstroDao"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  width="24"
                >
                  <use
                    href="/_next/undefined"
                  />
                </svg>
              </a>
              <a
                href="https://t.me/astro_near"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  width="24"
                >
                  <use
                    href="/_next/undefined"
                  />
                </svg>
              </a>
            </div>
            <div>
              <a
                href="https://github.com/near-daos/astro-ui/issues"
                rel="noreferrer noopener"
                target="_blank"
              >
                reportAnIssue
              </a>
            </div>
            <div>
              <a
                href="https://github.com/near-daos/astro-ui"
                rel="noreferrer noopener"
                target="_blank"
              >
                components.appFooter.githubRepo
              </a>
              <a
                href="https://testnet.app.astrodao.com/"
                rel="noreferrer noopener"
                target="_blank"
              >
                Testnet env
              </a>
              <a
                href="https://api.app.astrodao.com/docs/#/"
                rel="noreferrer noopener"
                target="_blank"
              >
                Swagger API
              </a>
            </div>
            <div>
              components.appFooter.opensourceAsIs
              <br />
              components.appFooter.communityDeveloped
            </div>
            <div>
              <a
                href="/terms-conditions"
              >
                termsAndConditions
              </a>
            </div>
            <div>
              <a
                href=""
                rel="noreferrer noopener"
                target="_blank"
              >
                releaseNotes
              </a>
            </div>
            <div>
              <div>
                <div>
                  <div
                    class=""
                  >
                    <img
                      data-testid="circle-country-flag"
                      height="100"
                      src="https://hatscripts.github.io/circle-flags/flags/us.svg"
                      title="us"
                    />
                     
                    en
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </aside>
    <div>
      <header>
        <a
          class=""
          href="https://astrodao.com/"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            width="100"
          >
            <use
              href="/_next/undefined"
            />
          </svg>
        </a>
        <div
          class=""
        >
          <a
            class=""
            href="https://astrodao.com/"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              width="100"
            >
              <use
                href="/_next/undefined"
              />
            </svg>
          </a>
          <div
            class=""
            style=""
          >
            <div>
              <button
                class=""
                type="button"
              >
                <svg>
                  <use
                    href="/_next/undefined"
                  />
                </svg>
              </button>
            </div>
            <input
              class="body1"
              placeholder="Search"
              tabindex="0"
              type="text"
              value=""
            />
            <div>
              <button
                class=""
                type="button"
              >
                <svg>
                  <use
                    href="/_next/undefined"
                  />
                </svg>
              </button>
            </div>
            <div />
          </div>
        </div>
        <div>
          <div>
            <button
              class=""
              type="button"
            >
              <svg
                class=""
                fill="none"
                height="38"
                viewBox="0 0 38 38"
                width="38"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="19"
                  cy="19"
                  r="18.5"
                />
                <path
                  d="M24.4384 11.0621L20.6769 16.5738C20.6159 16.6534 20.5877 16.7528 20.598 16.852C20.6083 16.9513 20.6562 17.043 20.7322 17.1087C20.8081 17.1745 20.9065 17.2094 21.0075 17.2065C21.1085 17.2035 21.2046 17.1629 21.2765 17.0929L24.978 13.9354C24.9994 13.916 25.026 13.9033 25.0546 13.8988C25.0832 13.8943 25.1125 13.8981 25.1389 13.9099C25.1652 13.9217 25.1876 13.9408 25.203 13.965C25.2185 13.9892 25.2264 14.0173 25.2258 14.0459V23.9703C25.2255 24.0004 25.2159 24.0297 25.1983 24.0543C25.1807 24.0789 25.1559 24.0976 25.1272 24.1079C25.0985 24.1182 25.0673 24.1196 25.0378 24.1119C25.0083 24.1043 24.9819 24.0879 24.962 24.065L13.7695 10.843C13.5908 10.6307 13.3673 10.4596 13.1145 10.3417C12.8618 10.2237 12.586 10.1618 12.3065 10.1602H11.9167C11.4084 10.1602 10.9208 10.3596 10.5614 10.7145C10.2019 11.0694 10 11.5508 10 12.0527V26.0286C10 26.5305 10.2019 27.0119 10.5614 27.3668C10.9208 27.7217 11.4084 27.9211 11.9167 27.9211C12.2443 27.921 12.5663 27.8381 12.8522 27.6802C13.1381 27.5224 13.3782 27.2948 13.5496 27.0192L17.3111 21.5074C17.3721 21.4279 17.4003 21.3285 17.39 21.2292C17.3797 21.13 17.3318 21.0383 17.2558 20.9725C17.1799 20.9068 17.0815 20.8718 16.9805 20.8748C16.8795 20.8777 16.7834 20.9183 16.7115 20.9884L13.01 24.1459C12.9887 24.1652 12.962 24.1779 12.9334 24.1825C12.9049 24.187 12.8755 24.1831 12.8492 24.1714C12.8228 24.1596 12.8005 24.1404 12.785 24.1163C12.7695 24.0921 12.7616 24.064 12.7622 24.0354V14.1209C12.7625 14.0907 12.7721 14.0614 12.7897 14.0368C12.8073 14.0122 12.8321 13.9935 12.8608 13.9832C12.8895 13.9729 12.9207 13.9715 12.9502 13.9792C12.9797 13.9869 13.0062 14.0032 13.026 14.0261L24.2185 27.2481C24.3986 27.4585 24.6229 27.6275 24.876 27.7434C25.129 27.8593 25.4046 27.9193 25.6835 27.9191H26.0833C26.335 27.9191 26.5842 27.8702 26.8168 27.775C27.0493 27.6799 27.2606 27.5405 27.4386 27.3648C27.6166 27.1891 27.7578 26.9804 27.8541 26.7508C27.9504 26.5212 28 26.2751 28 26.0266V12.0527C28 11.8032 27.95 11.5561 27.853 11.3257C27.7559 11.0953 27.6136 10.8862 27.4344 10.7103C27.2551 10.5344 27.0424 10.3952 26.8085 10.3008C26.5745 10.2064 26.324 10.1586 26.0713 10.1602C25.7437 10.1602 25.4217 10.2431 25.1358 10.401C24.8499 10.5589 24.6098 10.7865 24.4384 11.0621Z"
                />
              </svg>
            </button>
          </div>
        </div>
      </header>
      <div
        id="PAGE_LAYOUT_ID"
      >
        Hello World!
      </div>
      <div
        class=""
      >
        <button
          class=""
          type="button"
        >
          <svg>
            <use
              href="/_next/undefined"
            />
          </svg>
        </button>
      </div>
    </div>
    NotificationContainer
  </div>
</div>
`;
